test/pkg/framework/kafka.go

```diff
@@ -72,3 +76,89 @@ func ScaleKafkaInstanceDown(namespace, kafkaInstanceName string) error {
 
 	return nil
 }
+
+// WaitForMessagesOnTopic waits for at least a certain number of messages are present on the given topic
+func WaitForMessagesOnTopic(namespace, kafkaInstanceName, topic string, numberOfMsg int, timeoutInMin int) error {
+	return WaitForOnOpenshift(namespace, fmt.Sprintf("%d message available on topic %s withing %d minutes", numberOfMsg, topic, timeoutInMin), timeoutInMin,
+		func() (bool, error) {
+			messages, err := GetMessagesOnTopic(namespace, kafkaInstanceName, topic)
+			if err != nil {
+				return false, err
+			}
+			GetLogger(namespace).Info(fmt.Sprintf("Got %d messages", len(messages)))
+			return len(messages) >= numberOfMsg, nil
+		})
+}
+
+// GetMessagesOnTopic gets all messages for a topic
+func GetMessagesOnTopic(namespace, kafkaInstanceName, topic string) ([]string, error) {
+	kafkaInstance, err := GetKafkaInstance(namespace, kafkaInstanceName)
+	if err != nil {
+		return nil, err
+	}
+	GetLogger(namespace).Debug("Got kafka instance", "instance", kafkaInstance.Name)
+	bootstrapServer := infrastructure.ResolveKafkaServerURI(kafkaInstance)
+	if len(bootstrapServer) <= 0 {
+		GetLogger(namespace).Debug("Not able resolve URI for given kafka instance")
+		return nil, fmt.Errorf("not able resolve URI for given kafka instance %s", kafkaInstance.Name)
+	}
+	GetLogger(namespace).Debug("Got bootstrapServer", "server", bootstrapServer)
+
+	args := []string{"run", "-ti", "kafka-consumer", "--restart=Never"}
```

Comment:
Is it needed to start a new pod to connect to the bootstrap server? Can't we retrieve it from existing Kafka pod using `kubectl exec`?

Reply:
right ... gonna update

Reply:
updated !